pkg/hubble/observer/local_observer.go

```diff
@@ -606,7 +602,7 @@ func newRingReader(ring *container.Ring, req genericRequest, whitelist, blacklis
 	// correct index, then create a new reader that starts from there
 	for i := ring.Len(); i > 0; i, idx = i-1, idx-1 {
 		e, err := reader.Previous()
-		if errors.Is(err, container.ErrInvalidRead) {
```

Comment:
I think this is not correct: Invalid reads are only one source of lost events - the ring buffer will contain lost events for other reasons too [1], in which case we have _not_ reached the end of the ring buffer and need to continue rewinding.

We either need to check for  `LostEventSource_HUBBLE_RING_BUFFER` here. Or, alternatively, we could keep `ErrInvalidRead` and just translate replace `errors.New("requested data has been overwritten and is no longer available")` into a `LostEvent`.

[1] e.g. `PERF_EVENT_RING_BUFFER`, which means we dropped events in the kernel perf buffer before the reached Hubble. These events not very common, but they are stored in the ring buffer to indicate "some data is missing here due to userspace not keeping up with the kernel". If we observe them, we still want to continue rewinding.

Reply:
fixed!